### PR TITLE
refactored the Text Component to the size-steps according to design

### PR DESCRIPTION
**Pull request by robinvdv12**
### This PR:

The `Text` component had different size-steps then design had designed. This PR implements a new way to use the `Text` component, according to the design.

proposed release: `minor`

**Checklist** 🛡
- [x] I have exported my addition from `src/index.ts` (check if not applicable)
- [x] Appropriate tests have been added for my functionality (check if not applicable)

**Changes** 🌀
- Property `variant` is changed from: `small` | `base` | `large` to: `small` | `regular` | `large` | `XL` | `display`. Use this prop to set the font-size.
- Property `severity` accepts also `default` and `info` now. Use this prop to set the color.

**Removes** 👋
- `descriptive` property. If you want to make the text look descriptive now, just set the `severity`-property to `info`, like this: `<Text severity="info">Descriptive text</Text>`

### CODE REVIEW

File: src/components/Text/story.tsx [hunk 1]
```diff
@@ -31,13 +31,12 @@ const demoContent = `
 
 storiesOf('Text', module).add('Default', () => (
     <Text
-        strong={boolean('strong', false)}
-        descriptive={boolean('descriptive', false)}
+        variant={select('variant', ['small', 'regular', 'large', 'XL', 'display'], 'regular') as PropsType['variant']}
```
Comment (nicokoelewijn) [inline src/components/Text/story.tsx]:
```suggestion
        variant={select('variant', ['small', 'regular', 'large', 'extraLarge', 'display'], 'regular') as PropsType['variant']}
```

File: src/components/Text/style.tsx [hunk 2]
```diff
@@ -2,66 +2,65 @@ import _R from 'react';
...
-            return theme.Text.severity[severity].color;
+    }};
+    line-height: ${({ variant, compact, theme }): string => {
+        if (compact && variant) {
+            return theme.Text.variant[variant].lineHeight.compact;
+        } else if (!compact && variant) {
+            return theme.Text.variant[variant].lineHeight.default;
+        } else if (compact && !variant) {
+            return theme.Text.variant.regular.lineHeight.compact;
+        } else {
```
Comment (nicokoelewijn) [inline src/components/Text/style.tsx]: You can make this an early return (just remove the else statement)

File: src/components/Text/style.tsx [hunk 3]
```diff
@@ -2,66 +2,65 @@ import _R from 'react';
...
+    font-family: ${({ variant, theme }): string =>
+        !variant ? theme.Text.variant.regular.fontFamily : theme.Text.variant[variant].fontFamily};
+    font-size: ${({ variant, theme }): string =>
+        !variant ? theme.Text.variant.regular.fontSize : theme.Text.variant[variant].fontSize};
+    font-weight: ${({ variant, strong, theme }): string => {
+        if (strong) {
+            return theme.Text.strong.fontWeight;
+        } else if (variant && !strong) {
+            return theme.Text.variant[variant].fontWeight;
+        } else {
```
Comment (nicokoelewijn) [inline src/components/Text/style.tsx]: You can make this an early return (just remove the else statement)

File: src/themes/MosTheme/MosTheme.theme.ts [hunk 4]
```diff
@@ -697,32 +709,62 @@ const theme: ThemeType = {
...
+            large: {
+                fontFamily: bodyFont,
+                fontSize: '18px',
+                fontWeight: fontWeight.light,
+                lineHeight: {
+                    default: '27px',
+                    compact: '21px',
+                },
+            },
+            XL: {
```
Comment (nicokoelewijn) [inline src/themes/MosTheme/MosTheme.theme.ts]:
```suggestion
            extraLarge: {
```

File: src/components/Text/index.tsx [hunk 5]
```diff
@@ -1,16 +1,14 @@
 import React, { SFC } from 'react';
 import { StyledType } from '../../utility/styled';
 import { StyledSpan, StyledParagraph } from './style';
-import SeverityType from '../../types/SeverityType';
 
 type PropsType = StyledType & {
+    variant?: 'small' | 'regular' | 'large' | 'extraLarge' | 'display';
+    severity?: 'default' | 'error' | 'success' | 'info' | 'warning';
```
[the remark this answers is not in the document]
Comment (robinvdv12) [reply src/components/Text/index.tsx]: that's a great idea, thanks!

**Pull request merged**